components/openid-token-builder/src/main/java/com/wso2telco/openidtokenbuilder/CustomAuthorizationCodeGrantHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.wso2telco.openidtokenbuilder;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.AuthorizationCodeGrantHandler;
+
+import com.wso2telco.core.config.DataHolder;
+import com.wso2telco.ids.datapublisher.util.DataPublisherUtil;
+
+public class CustomAuthorizationCodeGrantHandler extends AuthorizationCodeGrantHandler {
+
+    @Override
+    public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+        boolean isvalidateGrant = super.validateGrant(tokReqMsgCtx);
+        boolean dataPublisherEnabled = DataHolder.getInstance().getMobileConnectConfig().getDataPublisher().isEnabled();
+        if (dataPublisherEnabled) {
+            String clientId = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getClientId();
+            Map<String, String> tokenMap = new HashMap<String, String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS2293)